src/dbos-executor.ts

```diff
@@ -1339,47 +1389,70 @@ export class DBOSExecutor implements DBOSExecutorContext {
       const wrappedProcedure = async (client: PoolClient): Promise<R> => {
         const ctxt = new StoredProcedureContextImpl(client, wfCtx, span, this.logger, funcId, proc.name);
 
+        let prevResult: R | Error | DBOSNull = dbosNull;
+        const queryFunc = <T>(sql: string, args: unknown[]) =>
+          this.procedurePool.query(sql, args).then((v) => v.rows as T[]);
         if (wfCtx.presetUUID) {
-          const func = <T>(sql: string, args: unknown[]) =>
-            this.procedurePool.query(sql, args).then((v) => v.rows as T[]);
-          const check: BufferedResult = await this.#checkExecution<R>(func, wfCtx.workflowUUID, funcId);
-          txn_snapshot = check.txn_snapshot;
-          if (check.output !== dbosNull) {
+          const executionResult = await this.#checkExecution<R>(queryFunc, wfCtx.workflowUUID, funcId);
+          prevResult = executionResult.result;
+          txn_snapshot = executionResult.txn_snapshot;
+          if (prevResult !== dbosNull) {
             ctxt.span.setAttribute('cached', true);
-            ctxt.span.setStatus({ code: SpanStatusCode.OK });
-            this.tracer.endSpan(ctxt.span);
-            return check.output as R;
+
+            if (this.debugMode === DebugMode.TIME_TRAVEL) {
+              // for time travel debugging, navigate the proxy to the time of this transaction's snapshot
+              await queryFunc(`--proxy:${executionResult.txn_id ?? ''}:${txn_snapshot}`, []);
+            } else {
+              // otherwise, return/throw the previous result
+              if (prevResult instanceof Error) {
```

Comment:
Same comment as above. Replay debug mode still need to make sure we have recorded output.

Reply:
Like above, this is in a `prevResult !== dbosNull` block and we throw `DBOSDebuggerError` if we're debugging and there's no previous result.